src/components/Dab.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+	<div 
+		class="wikibase-termbox-term" 
+		:class="modifier">
+		<div>
+			<span class="wikibase-termbox-term__language">{{ languageName }}</span>
+			<h2 class="wikibase-termbox-term__label">{{ label }}</h2>
+			<p class="wikibase-termbox-term__description">{{ description }}</p>
+			<ul v-if="hasAliases" class="wikibase-termbox-term__aliases">
+				<li v-for="alias in aliases"
+				class="wikibase-termbox-term__alias"
+				:data-separator="message( MESSAGE_KEYS.ALIAS_SEPARATOR )">{{ alias.value }}</li>
+			</ul>
+			<p class="wikibase-termbox-term__aliases wikibase-termbox-term__aliases--placeholder" v-else>?</p>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import Vue, { VueConstructor } from 'vue';
+import Component, { mixins } from 'vue-class-component';
+import Messages, { MessagesMixin } from './mixins/Messages';
+import {
+	mapState,
+	mapGetters,
+} from 'vuex';
+import {
+	NS_ENTITY,
+	NS_LANGUAGE,
+	NS_USER,
+} from '@/store/namespaces';
+import Term from '@/datamodel/Term';
+
+interface EntityBindings {
+	entityLabel: ( languageCode: string ) => Term;
+	entityDescription: ( languageCode: string ) => Term;
+	entityAliases: ( languageCode: string ) => Term[];
+}
+
+interface DabBindings extends Vue, EntityBindings, MessagesMixin {
+	language: string;
+	modifier: string;
+	getLanguageTranslation( language: string, inLanguage: string ): string;
+}
+
+@Component( {
+	props: {
+		language: {
+			type: String,
+			required: true,
+		},
+		modifier: {
+			type: String,
+			required: true,
+		},
+	},
+	computed: {
+		...mapState( NS_USER, [ 'primaryLanguage' ] ),
+		...mapGetters( NS_ENTITY, {
+			entityLabel: 'getLabelByLanguage',
+			entityDescription: 'getDescriptionByLanguage',
+			entityAliases: 'getAliasesByLanguage',
+		} ),
+		...mapGetters( NS_LANGUAGE, {
+			getLanguageTranslation: 'getTranslationByCode',
+		} ),
+	},
+} )
+export default class Dab extends ( mixins( Messages ) as VueConstructor<DabBindings> ) {
+	get label(): string {
+		const label: Term = this.entityLabel( this.language );
+		if ( label === null ) {
+			return '???';
+		} else {
+			return label.value;
+		}
+	}
+
+	get description(): string {
+		const description: Term = this.entityDescription( this.language );
+		if ( description === null ) {
+			return '??';
+		} else {
+			return description.value;
+		}
+	}
+
+	get hasAliases(): boolean {
+		return !( this.entityAliases( this.language ) == null );
+	}
+
+	get aliases(): Term[] {
+		const aliases: Term[] =  this.entityAliases( this.language );
+		if ( aliases === null ) {
+			return [];
+		} else {
+			return aliases;
+		}
+	}
+
+	get languageName(): string {
+		const name = this.getLanguageTranslation( this.language, this.primaryLanguage );
```

Comment:
The `primaryLanguage` property is proudly presented (only) by the `MessagesMixin`. Had a brief discussion with @jakobw about whether or not we regard this as desirable when we first introduced the mixin. Works but I don't like it.